configuration.c

```diff
@@ -174,14 +174,14 @@ getint(int c, int *int_r, gnc_t gnc, void *closure)
 static unsigned int
 getuint(int c, unsigned int *int_r, gnc_t gnc, void *closure)
 {
-    char *t, *end;
+    char *t = NULL, *end = NULL;
     unsigned int i;
     c = getword(c, &t, gnc, closure);
     if(c < -1)
         return c;
     errno = 0;
     i = strtoul(t, &end, 0);
-    if(*end != '\0' || errno) {
+    if(!t || t[0] == '-' || *end != '\0' || errno) {
```

Comment:
I'm out of the context but why we're downcasting `unsigned long` into `unsigned int`? Would it be possible to keep the type?